src/wp-includes/interactivity-api/interactivity-api.php

```diff
@@ -48,7 +48,9 @@ function wp_interactivity_process_directives_of_interactive_blocks( array $parse
 				list($root_block_name, $root_parsed_block) = $root_interactive_block;
 				if ( $root_block_name === $parsed_block['blockName'] && $parsed_block === $root_parsed_block ) {
 					// The root interactive blocks has finished rendering, process it.
-					$content = wp_interactivity_process_directives( $content );
+					if ( $content && str_contains( $content, 'data-wp-' ) ) {
+						$content = wp_interactivity_process_directives( $content );
```

Comment:
It might also make sense to move this check to the `wp_interactivity_process_directives` function as a good guard before using all the necessary logic to locate and process directives.

Reply:
Agree with that, it would be in line 108

Reply:
Well, would you want to add it to `wp_interactivity_process_directives()` or to `\WP_Interactivity_API::process_directives()`? Bit of a difference.

Reply:
`\WP_Interactivity_API::process_directives()` - that would be best, but we can follow up after 6.5 😅

Reply:
Addressed now